### PR TITLE
Reino/employee data service refactor

### PR DESCRIPTION
**Pull request by ReinoRabbit**
**Refactored the Employee Data Service**

- Added authorization checks
- Added checks to see if models exist
- Updated the unit tests

image
image

### CODE REVIEW

**Review by Juanro-Minne — APPROVED** [review]

**Review by madwizzard — APPROVED** [review]

**Review by EduanO — CHANGES_REQUESTED** [review]
Please go fix the logic of querying the data. We should not pull all data and then filter from them. 

Fix method naming.

Apply logical checks as shown in the diagram on the user story.

**Review by EduanO — APPROVED** [review]

File: HRIS.Services.Tests/Services/EmployeeDataServiceUnitTest.cs [hunk 1]
```diff
@@ -13,81 +15,260 @@ public class EmployeeDataServiceUnitTest
...
+    //        .ReturnsAsync(EmployeeDataTestData.EmployeeDataTwo); // Simulate adding new employee data
+
+    //    var employeeDataService = new EmployeeDataService(_dbMock.Object, mockIdentity.Object);
+
+    //    var result = await employeeDataService.CreateEmployeeData(newEmployeeDataDto);
+
+    //    Assert.NotNull(result);
+    //    Assert.Equivalent(EmployeeDataTestData.EmployeeDataTwo.ToDto(), result);
+    //    _dbMock.Verify(x => x.EmployeeData.Add(It.IsAny<EmployeeData>()), Times.Once);
+    //}
```
Comment (EduanO) [inline HRIS.Services.Tests/Services/EmployeeDataServiceUnitTest.cs]: Ghost code

**Review by EduanO — CHANGES_REQUESTED** [review]
Please look at using directives as well.

File: HRIS.Services.Tests/Services/EmployeeDataServiceUnitTest.cs (hunk 1 above)
Comment (ReinoRabbit) [reply HRIS.Services.Tests/Services/EmployeeDataServiceUnitTest.cs]: Removed ghost code

**Review by bseptember — APPROVED** [review]

**Review by EduanO — APPROVED** [review]

**Pull request merged**